src/multiplex/client.rs

```diff
@@ -259,12 +471,11 @@ where
                         tracing::trace!("request sent");
                         drop(guard);
 
-                        this.responses.push_back(Pending {
-                            tag: id,
-                            tx: res,
-                            span: _span,
-                        });
-                        this.in_flight.fetch_add(1, atomic::Ordering::AcqRel);
+                        pending.as_mut().sent(
+                            id,
+                            Pending::new(res, span, this.in_flight.clone()),
```

Comment:
This seems really unfortunate — we should have to clone an `Arc` for every single request. Can we instead put this tracking in the `PendingStore` maybe?